pkg/asset/installconfig/gcp/client.go

```diff
@@ -109,6 +116,28 @@ func (c *Client) GetPublicDNSZone(ctx context.Context, project, baseDomain strin
 	if res == nil {
 		return nil, errors.New("no matching public DNS Zone found")
 	}
+
+	return res, nil
+}
+
+// GetPublicDNSZoneByID returns a public DNS zone for a basedomain by its ID
+func (c *Client) GetPublicDNSZoneByID(ctx context.Context, project, id string) (*dns.ManagedZone, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
+	defer cancel()
+
+	svc, err := c.getDNSService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := svc.ManagedZones.Get(project, id).Context(ctx).Do()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get DNS Zone")
+	}
+	if res.Visibility == "private" {
+		return nil, errors.New("no matching public DNS Zone found")
+	}
```

Comment:
i think error should be, `$id must be public zone, but $visibility zone provided`

Reply:
Updated.